The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12508539112/job/34896789005) [(plain)](https://github.com/rust-lang/rust/commit/829667500e44e4d230ec4956eebaaa848c23b9ef/checks/34896789005/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.801 Building wheels for collected packages: reuse
#13 2.802   Building wheel for reuse (pyproject.toml): started
#13 3.052   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.053   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.053   Stored in directory: /tmp/pip-ephem-wheel-cache-190b6c0t/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.056 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.445 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.445 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.980 Collecting virtualenv
#13 3.980 Collecting virtualenv
#13 4.020   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.150      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 33.5 MB/s eta 0:00:00
#13 4.189 Collecting distlib<1,>=0.3.7
#13 4.194   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.203      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 69.7 MB/s eta 0:00:00
#13 4.239 Collecting filelock<4,>=3.12.2
#13 4.244   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.280 Collecting platformdirs<5,>=3.9.1
#13 4.284   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.366 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.551 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      131008 kB
DirectMap2M:     7208960 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
Diff in /checkout/src/librustdoc/clean/mod.rs:396:
     })
 }
 
-fn clean_region_outlives_predicate(
-    pred: ty::RegionOutlivesPredicate<'_>,
-) -> WherePredicate {
+fn clean_region_outlives_predicate(pred: ty::RegionOutlivesPredicate<'_>) -> WherePredicate {
     let ty::OutlivesPredicate(a, b) = pred;
     WherePredicate::RegionPredicate {
     WherePredicate::RegionPredicate {
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/src/librustdoc/clean/utils/tests.rs" "/checkout/src/librustdoc/clean/simplify.rs" "/checkout/src/librustdoc/clean/render_macro_matchers.rs" "/checkout/src/librustdoc/clean/inline.rs" "/checkout/src/librustdoc/clean/cfg/tests.rs" "/checkout/src/librustdoc/clean/cfg.rs" "/checkout/src/librustdoc/clean/utils.rs" "/checkout/src/librustdoc/clean/mod.rs" "/checkout/src/librustdoc/clean/types/tests.rs" "/checkout/src/librustdoc/clean/auto_trait.rs" "/checkout/src/librustdoc/clean/blanket_impl.rs" "/checkout/src/librustdoc/externalfiles.rs" "/checkout/src/librustdoc/doctest/make.rs" "/checkout/src/librustdoc/doctest/markdown.rs" "/checkout/src/librustdoc/doctest/runner.rs" "/checkout/src/librustdoc/doctest/tests.rs" "/checkout/src/librustdoc/doctest/rust.rs" "/checkout/src/librustdoc/json/import_finder.rs" "/checkout/src/librustdoc/json/conversions.rs" "/checkout/src/librustdoc/json/mod.rs" "/checkout/src/librustdoc/doctest.rs" "/checkout/src/librustdoc/docfs.rs" "/checkout/src/librustdoc/formats/item_type.rs" "/checkout/src/librustdoc/formats/cache.rs" "/checkout/src/librustdoc/formats/mod.rs" "/checkout/src/librustdoc/formats/renderer.rs" "/checkout/src/tools/tier-check/src/main.rs" "/checkout/src/tools/suggest-tests/src/dynamic_suggestions.rs" "/checkout/src/tools/suggest-tests/src/main.rs" "/checkout/src/tools/suggest-tests/src/static_suggestions.rs" "/checkout/src/tools/suggest-tests/src/lib.rs" "/checkout/src/tools/suggest-tests/src/tests.rs" "/checkout/src/tools/collect-license-metadata/src/path_tree.rs" "/checkout/src/tools/collect-license-metadata/src/licenses.rs" "/checkout/src/tools/collect-license-metadata/src/main.rs" "/checkout/src/tools/collect-license-metadata/src/reuse.rs" "/checkout/src/tools/error_index_generator/main.rs" "/checkout/src/tools/html-checker/main.rs" "/checkout/src/tools/remote-test-server/src/main.rs" "/checkout/src/tools/generate-copyright/src/cargo_metadata.rs" "/checkout/src/tools/generate-copyright/src/main.rs" "/checkout/src/tools/wasm-component-ld/src/main.rs" "/checkout/src/tools/llvm-bitcode-linker/src/target.rs" "/checkout/src/tools/llvm-bitcode-linker/src/bin/llvm-bitcode-linker.rs" "/checkout/src/tools/llvm-bitcode-linker/src/linker.rs" "/checkout/src/tools/llvm-bitcode-linker/src/lib.rs" "/checkout/src/tools/llvm-bitcode-linker/src/opt.rs" "/checkout/src/tools/run-make-support/src/string.rs" "/checkout/src/tools/run-make-support/src/assertion_helpers.rs" "/checkout/src/tools/run-make-support/src/util.rs" "/checkout/src/tools/run-make-support/src/scoped_run.rs" "/checkout/src/tools/run-make-support/src/env.rs" "/checkout/src/tools/run-make-support/src/external_deps/c_build.rs" "/checkout/src/tools/run-make-support/src/external_deps/c_cxx_compiler/extras.rs" "/checkout/src/tools/run-make-support/src/external_deps/c_cxx_compiler/gcc.rs" "/checkout/src/tools/run-make-support/src/external_deps/c_cxx_compiler/cc.rs" "/checkout/src/tools/run-make-support/src/external_deps/c_cxx_compiler/mod.rs" "/checkout/src/tools/run-make-support/src/external_deps/python.rs" "/checkout/src/tools/run-make-support/src/external_deps/clang.rs" "/checkout/src/tools/run-make-support/src/external_deps/cargo.rs" "/checkout/src/tools/run-make-support/src/external_deps/rustc.rs" "/checkout/src/tools/run-make-support/src/external_deps/mod.rs" "/checkout/src/tools/run-make-support/src/external_deps/llvm.rs" "/checkout/src/librustdoc/clean/types.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Thu Dec 26 21:30:58 UTC 2024
  network time: Thu, 26 Dec 2024 21:30:58 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/librustdoc/clean/mod.rs
+++ b/src/librustdoc/clean/mod.rs
@@ -268,7 +268,7 @@ fn clean_poly_trait_ref_with_constraints<'tcx>(
     )
 }
 
-fn clean_lifetime(lifetime: &hir::Lifetime, cx: &mut DocContext<'_>) -> Lifetime {
+fn clean_lifetime(lifetime: &hir::Lifetime, cx: &DocContext<'_>) -> Lifetime {
     if let Some(
         rbv::ResolvedArg::EarlyBound(did)
         | rbv::ResolvedArg::LateBound(_, _, did)
@@ -362,9 +362,9 @@ pub(crate) fn clean_predicate<'tcx>(
     let bound_predicate = predicate.kind();
     match bound_predicate.skip_binder() {
         ty::ClauseKind::Trait(pred) => clean_poly_trait_predicate(bound_predicate.rebind(pred), cx),
-        ty::ClauseKind::RegionOutlives(pred) => clean_region_outlives_predicate(pred),
+        ty::ClauseKind::RegionOutlives(pred) => Some(clean_region_outlives_predicate(pred)),
         ty::ClauseKind::TypeOutlives(pred) => {
-            clean_type_outlives_predicate(bound_predicate.rebind(pred), cx)
+            Some(clean_type_outlives_predicate(bound_predicate.rebind(pred), cx))
         }
         ty::ClauseKind::Projection(pred) => {
             Some(clean_projection_predicate(bound_predicate.rebind(pred), cx))
@@ -396,32 +396,30 @@ fn clean_poly_trait_predicate<'tcx>(
     })
 }
 
-fn clean_region_outlives_predicate(
-    pred: ty::RegionOutlivesPredicate<'_>,
-) -> Option<WherePredicate> {
+fn clean_region_outlives_predicate(pred: ty::RegionOutlivesPredicate<'_>) -> WherePredicate {
     let ty::OutlivesPredicate(a, b) = pred;
 
-    Some(WherePredicate::RegionPredicate {
+    WherePredicate::RegionPredicate {
         lifetime: clean_middle_region(a).expect("failed to clean lifetime"),
         bounds: vec![GenericBound::Outlives(
             clean_middle_region(b).expect("failed to clean bounds"),
         )],
-    })
+    }
 }
 
 fn clean_type_outlives_predicate<'tcx>(
     pred: ty::Binder<'tcx, ty::TypeOutlivesPredicate<'tcx>>,
     cx: &mut DocContext<'tcx>,
-) -> Option<WherePredicate> {
+) -> WherePredicate {
     let ty::OutlivesPredicate(ty, lt) = pred.skip_binder();
 
-    Some(WherePredicate::BoundPredicate {
+    WherePredicate::BoundPredicate {
         ty: clean_middle_ty(pred.rebind(ty), cx, None, None),
         bounds: vec![GenericBound::Outlives(
             clean_middle_region(lt).expect("failed to clean lifetimes"),
         )],
         bound_params: Vec::new(),
-    })
+    }
 }
 
 fn clean_middle_term<'tcx>(
@@ -1860,7 +1858,7 @@ pub(crate) fn clean_ty<'tcx>(ty: &hir::Ty<'tcx>, cx: &mut DocContext<'tcx>) -> T
 
 /// Returns `None` if the type could not be normalized
 fn normalize<'tcx>(
-    cx: &mut DocContext<'tcx>,
+    cx: &DocContext<'tcx>,
     ty: ty::Binder<'tcx, Ty<'tcx>>,
 ) -> Option<ty::Binder<'tcx, Ty<'tcx>>> {
     // HACK: low-churn fix for #79459 while we wait for a trait normalization fix

